src/workerd/api/urlpattern-standard.c++

```diff
@@ -68,16 +68,21 @@ ada::url_pattern_init URLPattern::URLPatternInit::toAdaType() const {
 }
 
 URLPattern::URLPatternInit URLPattern::createURLPatternInit(const ada::url_pattern_init& other) {
+  // By converting to USVString we are asserting that these values are always valid UTF-8 data,
+  // with no unpaired surrogates, etc. This is true because these values are derived from user
+  // input that was already a USVString, and ada-url will not introduce invalid UTF-8 data when
+  // it processes input.
+
   URLPatternInit result{};
 #define V(_, name)                                                                                 \
   if (auto v = other.name) {                                                                       \
-    result.name = kj::str(kj::ArrayPtr(v->c_str(), v->size()));                                    \
+    result.name = jsg::USVString(kj::str(kj::ArrayPtr(v->c_str(), v->size())));                    \
```

Comment:
Hmm am I supposed to be using @jasnell's accountedUSVString methods here?

Reply:
That would be preferred yes.